paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FullChangelogMergeFunctionWrapper.java

```diff
@@ -110,8 +112,9 @@ public ChangelogResult getResult() {
             } else {
                 if (merged == null || !isAdd(merged)) {
                     reusedResult.addChangelog(replace(reusedBefore, RowKind.DELETE, topLevelKv));
-                } else if (!changelogRowDeduplicate
-                        || !valueEqualiser.equals(topLevelKv.value(), merged.value())) {
+                } else if (!isFirstRow
```

Comment:
Do we have more intuitive modifications for `FullChangelogMergeFunctionWrapper` and `LookupChangelogMergeFunctionWrapper`?

If it is a first row, we should just return old value. And produce no changelog.
Can we just have another wrapper? Maybe...

Reply:
I have not come up with a good wrapper to handle this, but I improve the logic in `FullChangelogMergeFunctionWrapper` with a short-cut for first row